docs/src/getting_started/using_cmaize_as_build_system/cmake_like_project.rst

```diff
@@ -0,0 +1,174 @@
+of the project.
+
+.. code-block:: cmake
+
+    # Declare minimum version required
+    cmake_minimum_required(VERSION 3.13.4)
+
+    # Declare project name and version
+    project(MyCMakeProject VERSION 1.0)
+
+    # Set C++ standard
+    set(CMAKE_CXX_STANDARD 14)
+
+Including CMaize
+^^^^^^^^^^^^^^^^
+
+We also need to include CMaize. This is described more in
+:ref:`Automatically Downloading and Including CMaize`.
+
+.. code-block:: cmake
+
+    # Include CMaize automatically
+    include("${PROJECT_SOURCE_DIR}/cmake/get_cpp.cmake")
+
+Adding the TravelCostUtils Library
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now we will add our ``TravelCostUtils`` library. We need to specify the target
+name, its source directory, and its include directory:
+
+.. code-block:: cmake
+
+    # Add our TravelCostUtils library
+    cpp_add_library(
+        TravelCostUtils
+        SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/TravelCostUtils"
+        INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}/TravelCostUtils"
+    )
+
+Adding the TravelCostCalc Executable
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here we add our ``TravelCostCalc`` executable. We simply need to specify
+the target name, its source directory, and its dependencies:
+
+.. code-block:: cmake
+
+    # Add the executable
+    cpp_add_executable(
+        TravelCostCalc
+        SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/TravelCostCalc"
+        DEPENDS TravelCostUtils
+    )
+
+Getting the Catch2 Unit Test Framework
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now we will find and build the Catch2 unit testing framework. We need to do the
+following:
+
+#. Create a target name for it (``Catch2`` in this case)
+#. Point to its GitHub repository URL using the ``URL`` keyword
+#. Specify the target we want to build in the repository using the
+   ``BUILD_TARGET`` keyword
+#. Specify the target name we want to use to find this library using the
+   ``FIND_TARGET`` keyword
+#. Additionally we will pass in the ``BUILD_TESTING=OFF`` argument so that no
+   tests are built for the library. Passing in arguments is accomplished by
+   using the ``CMAKE_ARGS`` keyword.
+
+.. code-block:: cmake
+
+    # Get the Catch2 unit testing framework
+    cpp_find_or_build_dependency(
+        Catch2
+        URL github.com/catchorg/Catch2
+        BUILD_TARGET Catch2
+        FIND_TARGET Catch2::Catch2
+        CMAKE_ARGS BUILD_TESTING=OFF
+    )
+
+Adding Tests
+^^^^^^^^^^^^
+
+Now we can add our tests. We just need to specify the source directory and the
+tests dependencies (which in this case are our ``TravelCostUtils`` library and
+the ``Catch2`` testing framework).
+
+.. code-block:: cmake
+
+    # Add the tests
+    cpp_add_tests(
+        TravelCostUtilsTest
+        SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/test"
+        DEPENDS TravelCostUtils Catch2::Catch2
+    )
+
+.. note::
+
+    Here we are using ``Catch2::Catch2`` to find the Catch2 library. This is
+    what we specified as the ``FIND_TARGET`` when finding and building Catch2.
```

Comment:
It's not just that we called it `Catch2::Catch2` it's that Catch2 calls their exported target `Catch2::Catch2`. I guess my point is the names are not arbitrary, which is not clear from the wording.

Reply:
I updated this note but I'm still not sure that I'm getting the right message across.

Reply:
Don't worry about it too much since it'll probably go away with the superbuilds.